avro/src/error.rs

```diff
@@ -377,6 +377,8 @@ pub enum Error {
     #[error("Failed to compress with flate: {0}")]
     DeflateCompress(#[source] std::io::Error),
 
+    // no longer possible after migration from libflate to miniz_oxide
+    // TODO: remove in the next semver-breaking release
```

Comment:
Please add '#[deprecated(...)]'

Reply:
Done. I see the version in git is 0.18, while the latest version on crates.io is 0.17. That is a semver-incompatible version bump. Should I go ahead and remove the error entirely? Or are you planning on keeping compatibility with 0.17.x for a while?

Reply:
Despite still being 0.x we prefer to make it easier for the users. 
Keeping this error variant around for some months is not a problem!